feat(am6l2x): Add Documentation for TI LVGL Demo (#174)
TexasInstruments/processor-sdk-doc: source/system/Demo_User_Guides/TI_LVGL_Demo_User_Guide.rst

Reviewer: The entire documentation doesn't seem to follow sphinx guidelines for section headers - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections 

Please plan to go through the above guidelines & update each section header as per it

Author: Resolved

.. _TI-LVGL-Demo-User-Guide-label:

TI LVGL Demo - User Guide

> =============================

Author: Resolved

Overview
--------

TI LVGL Demo is a simple app that showcases the capabilities of the `LVGL - Light and Versatile Graphics Library <https://lvgl.io/>`__.
It includes various demo applications, such as:

- EV Charging

- Smart Home

- Smart Meter

- Thermostat

- Security

All necessary equipment and step by step instructions are provided below:

.. note::

   Some of the features within each of these applications are not supported on all platforms. So there will be differences in feature
   enablement across platforms.

Hardware Prerequisites
----------------------

.. ifconfig:: CONFIG_part_variant in ('AM62LX')

   -  AM62L Evalaution Module: TMDS62LEVM

> .. ifconfig:: CONFIG_part_variant in ('AM62X')

Reviewer: Good that you handled other platforms!
In that case, can you update commit message title - `feat: Add Documentation for TI LVGL Demo`

Author: Resolved

      By default CC33xx is configured at boot, so here are the steps that can be followed to enable it:

      .. code-block:: console

         $ systemctl stop ti-lvgl-demo
         $ cd /usr/share/cc33xx
         $ ./sta_start.sh
         $ ./sta_connect.sh -s WPA-PSK -n <SSID> -p <PASSWORD>
         $ udhcpc -i wlan0
         $ systemctl start ti-lvgl-demo

      For more details on how to enable CC33xx and connect to WiFi, visit :ref:`How to Enable M.2-CC33x1 in Linux <enable_m2cc3301>`

**As a part of this application, we have the following widgets:**

Widget#1: Climate Control
"""""""""""""""""""""""""

 .. note::

      This feature will work on platforms that have on-chip ADC. Following is supported on AM62Lx

 - The indoor temperature in this widget displays the digital data obtained from ADC and display (data/10) on the widget.
 - The outdoor temperature below the indoor teperature shows the value of (data/10)-6.5, if it is positive, else, shows zero.
 - The indoor temperature is sent over SSL-encrypted MQTT messages under the topic `SmartHome/temp`
 - For this demo, we are using channel 0 for ADC input. To change the channel being used, modify the following in

>    `ti-lvgl-demo/lv_port_linux/lvgl/demos/high_res/adc.c <https://github.com/TexasInstruments/ti-lvgl-demo.git/>`__

Reviewer: Better if the external hyperlink directly points to the `adc.c` file

Building the TI LVGL Demo
-------------------------

The TI LVGL Demo is enabled by default in yocto for |__PART_FAMILY_DEVICE_NAMES__| Linux. Note, that the binary itself does not have
asset images and slides built in it. :file:`tisdk-default-image` contains the required assets within :file:`/usr/share/ti-lvgl-demo/*`.
You will need to place any additional assets here while making any modifications. Yocto recipe for building this demo can be found at
`github: ti-lvgl-demo.bb <https://github.com/TexasInstruments/meta-tisdk/blob/scarthgap/recipes-demos/ti-lvgl-demo/ti-lvgl-demo.bb>`__

The source code is available at `TI LVGL Demo <https://github.com/TexasInstruments/ti-lvgl-demo.git/>`__ and can be re-compiled with the
following steps:

1. First clone the git repository and its submodules using:

   .. code-block:: console

      $ git clone --recurse-submodules https://github.com/TexasInstruments/ti-lvgl-demo.git

2. Create the docker environment and build the application:

   .. code-block:: console

      $ cd ti-lvgl-demo
      $ sudo ./scripts/docker_setup.sh --create-image
      $ sudo ./scripts/docker_setup.sh --build-app

> 3. Copy the compiled binary to /usr/bin directory of the device

Reviewer: ```suggestion
3. Copy the compiled binary to :file:`/usr/bin` directory of the device
```
role `file` can be used to highlight name of directory as well 

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-file

Author: Resolved

.. _TI-LVGL-Demo-User-Guide-label:

TI LVGL Demo - User Guide

> =========================

Reviewer: ```
#########################
TI LVGL Demo - User Guide
#########################
```

This should be treated as parts - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections
`# with overline, for parts`

Author: Resolved

Overview

> --------

Reviewer: This should be treated as chapter - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

```suggestion
********
Overview
********
```

`* with overline, for chapters`

Author: Resolved

TI LVGL Demo is a simple app that showcases the capabilities of the `LVGL - Light and Versatile Graphics Library <https://lvgl.io/>`__.
It includes various demo applications, such as:

- EV Charging

- Smart Home

- Smart Meter

- Thermostat

- Security

All necessary equipment and step by step instructions are provided below:

.. note::

   Some of the features within each of these applications are not supported on all platforms. So there will be differences in feature
   enablement across platforms.

Hardware Prerequisites

> ----------------------

Reviewer: This should be treated as chapter - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

Author: Resolved

.. ifconfig:: CONFIG_part_variant in ('AM62LX')

>    -  AM62L Evalaution Module: TMDS62LEVM

Reviewer: ```suggestion
   -  AM62L Evaluation Module: TMDS62LEVM
```

Author: Resolved

.. ifconfig:: CONFIG_part_variant in ('AM62LX')

   -  AM62L Evaluation Module: TMDS62LEVM

.. ifconfig:: CONFIG_part_variant in ('AM62X')

   -  TI AM62x SK / TI AM62x-LP SK / TI AM62xSIP SK / BeaglePlay

> .. ifconfig:: CONFIG_part_variant in ('AM62P')

Reviewer 2: ```suggestion
.. ifconfig:: CONFIG_part_variant in ('AM62PX')
```
Checkout - https://github.com/TexasInstruments/processor-sdk-doc/blob/master/configs/AM62PX/AM62PX_linux_config.txt#L38

Author: Resolved

-  PC (Windows or Linux, to use serial terminal console)

> -  HDMI Monitor (to view the Demo on Display)

Author: Resolved

-  Ethernet Cable or CC33xx WiFi module (to connect |__PART_FAMILY_DEVICE_NAMES__| SK to internet)

-  Mouse or Touch Input from HDMI monitor (to control the TI LVGL Demo)

-  SD card (minimum 16GB)

-  Aux cable connected to earphone/speaker

.. ifconfig:: CONFIG_part_variant in ('AM62LX')

   -  Jumper cable (to connect potentiometer to ADC input) or some analog signal connected to ADC input. In the J11 header, pin D7 (A01)

>       is the potentiometer out, which needs to be connected to pin D6 (A02) ADC in for channel 0.

Reviewer: @div2779 Do you want to link to the document from where you got this input?

Is it TRM?

 https://www.ti.com/lit/ug/sprujb4/sprujb4.pdf

Author: The document is from the EVM schematics, will see if we can get find the URL for the same

Author: Resolved. Schematics link added for AM62L.

      .. Image:: /images/ti-lvgl-demo-potentiometer-connection.png
         :height: 300

Launching the TI LVGL Demo

> --------------------------

Reviewer: This should be treated as chapter - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

Author: Resolved

The demo will auto launch upon Linux booting on the EVM. Follow the below instructions to flash the SD card:

1. Flash an SD card with the :file:`tisdk-default-image`. User can download the :file:`tisdk-default-image` wic image from |__SDK_DOWNLOAD_URL__|.
   Please follow the instructions from here to :ref:`Flash an SD card <processor-sdk-linux-create-sd-card>`.

.. note::

>    Images may vary between each platform based on the number of supported applications.

Reviewer: @div2779 what's the significance of this note?

Author: Note removed

> 2. Insert the flashed SD card to the board, connect the display, mouse/touch-input, ethernet cable, aux cable, jumper wire and power on the SK-EVM.

Reviewer: ```suggestion
2. Insert the flashed SD card to the board, connect the display, mouse/touch-input, ethernet cable, aux cable, jumper wire and power on the EVM.
```

As the line is generic and SK won't be appropriate for AM62L

Author: Resolved

   The TI LVGL Demo will launch automatically when the device is fully booted.

Using the TI LVGL Demo
----------------------

   The landing/home page of the LVGL demo looks like the following:

   .. Image:: /images/ti-lvgl-demo-home-page.gif
      :height: 500

   |

   - In the demo, scroll through the various widgets to launch different apps.

>    - The date/time panel shows the UTC timezone (if device is connected to the internet)

Reviewer: ```suggestion
   - The date/time panel shows the UTC timezone provided the EVM is connected to the internet
```

Author: Resolved

>    - Exit button on top left can be used to exit the demo. You can also use :file:`systemctl stop ti-lvgl-demo` to stop the demo.

Reviewer: File role is limited to names of files or directories.
In this case, as you are mentioning a systemctl command you should use code directive

```suggestion
   - Exit button on top left can be used to exit the demo. You can also use :code:`systemctl stop ti-lvgl-demo` to stop the demo.
```

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-code

Author: Resolved

>    - Sytem button (Center button in top-right toolbar) shows FPS and CPU Utilization of the Demo.

Reviewer: CPU utilization is still of the demo?
I thought it was for entire system
```suggestion
   - Sytem button (Center button in top-right toolbar) shows FPS and CPU Utilization of the system
```

>    - Sytem button (Center button in top-right toolbar) shows FPS and CPU Utilization of the Demo.

Author: Resolved

>    - You can switch between light annd dark theme from theme buttons in bottom-right toolbar.

Reviewer: ```suggestion
   - You can switch between light and dark theme from theme buttons in bottom-right toolbar.
```

Author: Resolved

     locations in `ti-lvgl-demo/lv_port_linux/lvgl/demos/high_res/ <https://github.com/TexasInstruments/ti-lvgl-demo.git/>`__:

      .. code-block:: c

         /* In  on_connect_vol_sub() function in mqtt_sub.c*/
         rc = mosquitto_subscribe(mosq, NULL, "SmartHome/volume", 1);
         :
         rc = mosquitto_subscribe(mosq, NULL, "SmartHome/led", 1);

         /* In  publish_sensor_data() function in adc.c*/
         rc = mosquitto_publish(mosq, NULL, "SmartHome/temp", strlen(payload), payload, 1, false);

         /* In  publish_evCharge_data() function in adc.c*/
         rc = mosquitto_publish(mosq, NULL, "SmartHome/evCharge", strlen(payload), payload, 1, false);

   .. note::

      By default CC33xx is configured at boot, so here are the steps that can be followed to enable it:

      .. code-block:: console

         $ systemctl stop ti-lvgl-demo
         $ cd /usr/share/cc33xx
         $ ./sta_start.sh
         $ ./sta_connect.sh -s WPA-PSK -n <SSID> -p <PASSWORD>
         $ udhcpc -i wlan0
         $ systemctl start ti-lvgl-demo

      For more details on how to enable CC33xx and connect to WiFi, visit :ref:`How to Enable M.2-CC33x1 in Linux <enable_m2cc3301>`

> 

Reviewer 3: As per https://github.com/TexasInstruments/processor-sdk-doc/pull/171 it should be CC3351 ( @sabeeh-khan  please comment)

Author: Since this PR is already merged and the page heading still contains M.2-CC33x1 instead of M.2-CC3351, let us let it stay as it until next release.

Widget#1: Climate Control
"""""""""""""""""""""""""

 .. note::

      This feature will work on platforms that have on-chip ADC. Following is supported on AM62Lx

 - The indoor temperature in this widget displays the digital data obtained from ADC and display (data/10) on the widget.

>  - The outdoor temperature below the indoor teperature shows the value of (data/10)-6.5, if it is positive, else, shows zero.

Reviewer: ```suggestion
 - The outdoor temperature below the indoor temperature shows the value of (data/10)-6.5, if it is positive, else, shows zero.
```

Author: Resolved

   - In HMI, the slider `Light temperature` is not associated with any functionality but can be programmed be user to perform any function.

Widget#6: Sensor Controls
"""""""""""""""""""""""""

   - The buttons/sliders in this widget are not associated with any functionality but can be programmed by the user to perform any function.

Launching the Smart Meter HMI
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To launch the Smart Meter HMI demo, click on the `Smart Meter` widget in the apps scroll menu.

   .. Image:: /images/ti-lvgl-demo-smart-meter.png
      :height: 300

Launching the Thermostat HMI
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To launch the Thermostat HMI demo, click on the `Thermostat` widget in the apps scroll menu.

   .. Image:: /images/ti-lvgl-demo-thermostat.png
      :height: 300

Launching the Security HMI
^^^^^^^^^^^^^^^^^^^^^^^^^^

To launch the Security HMI demo, click on the `Security` widget in the apps scroll menu.

   .. Image:: /images/ti-lvgl-demo-security.png
      :height: 300

This widget contains a slide-show on Application Processor Security for AM6X devices.

> Building the TI LVGL Demo

Reviewer: ```suggestion
Building the TI LVGL Demo from sources
```

Author: Resolved

-------------------------

> The TI LVGL Demo is enabled by default in yocto for |__PART_FAMILY_DEVICE_NAMES__| Linux. Note, that the binary itself does not have

Reviewer: ```suggestion
The TI LVGL Demo is enabled in :file:`tisdk-default-image` yocto filesystem for |__PART_FAMILY_DEVICE_NAMES__| by default. Note, that the binary itself does not have
```

Author: Resolved

asset images and slides built in it. :file:`tisdk-default-image` contains the required assets within :file:`/usr/share/ti-lvgl-demo/*`.
You will need to place any additional assets here while making any modifications. Yocto recipe for building this demo can be found at
`github: ti-lvgl-demo.bb <https://github.com/TexasInstruments/meta-tisdk/blob/scarthgap/recipes-demos/ti-lvgl-demo/ti-lvgl-demo.bb>`__

The source code is available at `TI LVGL Demo <https://github.com/TexasInstruments/ti-lvgl-demo.git/>`__ and can be re-compiled with the
following steps:

1. First clone the git repository and its submodules using:

   .. code-block:: console

      $ git clone --recurse-submodules https://github.com/TexasInstruments/ti-lvgl-demo.git

2. Create the docker environment and build the application:

   .. code-block:: console

      $ cd ti-lvgl-demo
      $ sudo ./scripts/docker_setup.sh --create-image
      $ sudo ./scripts/docker_setup.sh --build-app

3. Copy the compiled binary to :file:`/usr/bin` directory of the device

   .. code-block:: console

      $ scp lv_port_linux/bin/lvglsim root@<ip-addr-of-device>:/usr/bin/

> 

Reviewer: commands to transfer assests like image and slides should also be mentioned here

Author: Resolved
